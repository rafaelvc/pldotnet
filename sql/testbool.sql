-- Float4 (real): 6 digits of precison
CREATE OR REPLACE FUNCTION returnBool() RETURNS boolean AS $$
return false;
$$ LANGUAGE pldotnet;
SELECT returnBool();

