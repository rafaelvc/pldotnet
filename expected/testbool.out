CREATE OR REPLACE FUNCTION returnBool() RETURNS boolean AS $$
return false;
$$ LANGUAGE pldotnet;
SELECT returnBool() is false;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanAnd(a boolean, b boolean) RETURNS boolean AS $$
return a&b;
$$ LANGUAGE pldotnet;
SELECT BooleanAnd(true, true) is true;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanOr(a boolean, b boolean) RETURNS boolean AS $$
return a|b;
$$ LANGUAGE pldotnet;
SELECT BooleanOr(false, false) is false;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanXor(a boolean, b boolean) RETURNS boolean AS $$
return a^b;
$$ LANGUAGE pldotnet;
SELECT BooleanXor(false, false) is false;
 ?column? 
----------
 t
(1 row)

